#158652806 feature(requests): handle ride request response

CEOehis opened the pull request:
### What does this PR do?
feature(requests) handle ride request response

#### Description of Task to be completed?
- add method to handle ride request response
- handle 'accepted' and 'rejected' status
- prevent user from responding to another users' ride requests
- include appropriate tests

#### How should this be manually tested?
make a `PUT` request to `/rides/:rideId/requests/:requestId` with payload

#### Any background context you want to provide?
Users can use this endpoint to accept or reject a ride offer request

#### What are the relevant pivotal tracker stories?
[#158652806](https://www.pivotaltracker.com/story/show/158652806)
[#158652837](https://www.pivotaltracker.com/story/show/158652837)

CEOehis committed:
feature(requests) handle ride request response

- add method to handle ride request response
- prevent user from responding to another users ride requests
- include appropriate tests

[Delivers #158652806]

williamolojede commented on server/controllers/rideRequestController.js:
```diff
@@ -78,4 +87,62 @@ export default class RideRequest {
+   * @memberof RideRequest
+   */
+  static respondToRequests(req, res) {
+    const { status } = req.body;
```
you can combine lines 100 - 103 into one using object destructuring.

williamolojede commented on server/controllers/rideRequestController.js:
```diff
@@ -78,4 +87,62 @@ export default class RideRequest {
+    pool
+      .query('SELECT * FROM rides WHERE id = $1', [rideId])
+      .then((result) => {
+        if (result.rowCount) {
```
I understand that 0 is falsy in javascript but i'd prefer that you are explicit about what this if statement is checking, something like: 
```js
if(result.rowCount != 0) {
 ....
}
```

williamolojede commented on server/controllers/rideRequestController.js:
```diff
@@ -78,4 +87,62 @@ export default class RideRequest {
+    // check if the user is the one who created the ride offer
+    pool
+      .query('SELECT * FROM rides WHERE id = $1', [rideId])
+      .then((result) => {
```
you can use object destructuring here and also rename `rows` to rides`

CEOehis replied:
Ok. Thanks. 
I'll complete this in my next commit.

williamolojede commented on server/controllers/rideRequestController.js:
```diff
@@ -78,4 +87,62 @@ export default class RideRequest {
+        // could not retrieve ride
+        return res.status(500).json({
+          status: 'error',
+          message: error,
```
You shouldn't send auto-generated errors to the consumer of your api because you have no control over what the error message would be and the consumer of your api might rely on its consistency.

williamolojede commented on server/routes/index.js:
```diff
@@ -16,5 +16,7 @@ router.get('/rides/:id', Ride.getRideOffer);
 router.get('/users/rides/:rideId/requests', RideRequest.getRideRequests);
 router.post('/users/rides', validate.rideOfferValidator, Ride.createRideOffer);
 router.post('/rides/:rideId/requests', RideRequest.createRequest);
+router.put('/users/rides/:rideId/requests/:requestId', RideRequest.respondToRequests);
```
this should be `/rides/:rideId/requests/:requestId`

CEOehis replied:
Noted. Thank you.

CEOehis committed:
refactor(requests): make conditionals more explicit

- update route url
- refactor if statements to be more explicit
- use object destructuring

CEOehis committed:
feature(requests): check if request has been responded to

- check if status was provided in request body
- compare created_at and updated_at timestamps to check if request
has been responded to before
